frontend/packages/dev-console/integration-tests/tests/git-import-flow.scenario.ts

```diff
@@ -34,27 +64,219 @@ describe('git import flow', () => {
+    expect(appName.getAttribute('value')).toContain(newApp);
+    expect(getText(gitUrlHelper)).toBe('Validated');
+    await browser.wait(
+      until.visibilityOf(builderImageSelected),
+      WAIT,
+      `BuilderImage is not selected even after ${WAIT} milliseconds`,
+    );
+    expect(await builderImageSelected.isDisplayed()).toBe(true);
+
+    await browser.wait(
+      until.elementToBeClickable(createButton),
+      WAIT,
+      `Create button is not enabled even after ${WAIT} milliseconds`,
+    );
+    expect(createButton.isEnabled());
+    await createButton.click();
+    await browser.wait(until.urlContains(`${appHost}/topology/ns/${testName}`));
+    expect(browser.getCurrentUrl()).toContain(`${appHost}/topology/ns/${testName}`);
+  });
+});
+
+describe('git import flow with advanced options', () => {
+  let newApplication;
+  let newApp;
+
+  beforeAll(async () => {
+    await browser.get(`${appHost}/k8s/cluster/projects/${testName}`);
+    newApplication = newApplicationName();
+    newApp = newAppName();
+  });
+
+  afterEach(() => {
+    checkLogs();
+    checkErrors();
+  });
+
+  it('add git repo details', async () => {
+    await addNavigate.click();
+    await switchPerspective(Perspective.Developer);
+    expect(sideHeader.getText()).toContain('Developer');
+    await navigateImportFromGit();
+    await browser.wait(until.textToBePresentInElement(importFromGitHeader, 'Import from git'));
+    expect(importFromGitHeader.getText()).toContain('Import from git');
+    await enterGitRepoUrl(testData.gitRepoUrl);
+
+    await click(appName);
+    expect(appName.getAttribute('value')).toContain('nodejs-ex-git');
+    await addApplicationInGeneral(newApplication, newApp);
+    expect(await applicationName.getAttribute('value')).toContain(newApplication);
+    expect(await appName.getAttribute('value')).toContain(newApp);
+    expect(await getText(gitUrlHelper)).toBe('Validated');
+    expect(await builderImageSelected.isDisplayed()).toBe(true);
+  });
+
+  it('add Adanced option "Routing" details', async () => {
+    // Add Advanced option - Routing
+    await selectAdvancedOptions(AdvancedOptions.Routing);
+    await setRouting(testData.hostName, testData.path);
+    await setSecureRoute(TLSTerminationValues.Edge);
+    expect(routingObj.certificate.isDisplayed()).toBe(true);
+    expect(routingObj.caCertificate.isDisplayed()).toBe(true);
+    expect(routingObj.privateKey.isDisplayed()).toBe(true);
+  });
+
+  it('add Adanced option "BuildConfig" details', async () => {
+    // Add Advanced option - Build Config
+    await selectAdvancedOptions(AdvancedOptions.BuildConfig);
+    expect(await buildConfigObj.webHookBuildTrigger.getAttribute('value')).toBe('true');
+    expect(await buildConfigObj.buildTriggerImage.getAttribute('value')).toBe('true');
+    expect(await buildConfigObj.buildTriggerConfigField.getAttribute('value')).toBe('true');
+    await setBuildConfig(testData.buildEnvName, testData.buildEnvValue);
+  });
+
```

Comment:
Can we convert it to functions so others can use them?

Reply:
setBuildConfig is already a function with 2 params as environment name and environmental value

Reply:
I am talking about the whole advanced options. A different function for each advanced option. Not only, build config but the others as well.